src/posix/main.c

```diff
@@ -314,10 +314,17 @@ static otInstance *InitInstance(PosixConfig *aConfig)
     syslog(LOG_INFO, "Thread version: %hu", otThreadGetVersion());
     IgnoreError(otLoggingSetLevel(aConfig->mLogLevel));
 
-    instance = otSysInit(&aConfig->mPlatformConfig);
-
     atexit(otSysDeinit);
 
+    if (aConfig->mIsDryRun)
+    {
+        instance = otSysInitMinimal(&aConfig->mPlatformConfig);
```

Comment:
What if we add another option to get information only available when another component is initialized?

Reply:
For extension in the future, I think we can calculate which components should be initialized in this function from the config. Thoughts? @bukepo